### PR TITLE
[2315] part 3, Ignore redundant decline_by_default columns

### PR DESCRIPTION
**Pull request by elceebee**
## Context

This is the third PR to address a bug where `decline_by_default_at` and `decline_by_default_days` were being set inconsistently. See the [first PR here](https://github.com/DFE-Digital/apply-for-teacher-training/pull/9861) for context and then [the second](https://github.com/DFE-Digital/apply-for-teacher-training/pull/9875) to see what else has been done.

## Changes proposed in this pull request
- Some final clean up, removing some uses of these fields
- ignoring these columns
- ignoring the columns has some impact on the raw sql we use for sorting, so there are some changes there as well.

We will ignore these to columns and stop syncing them with BigQuery. This approach has been confirmed with the data team. 

We are _not_ deleting the columns because the data is still relevant pre-continuous applications when the dates were used and the decline by default job was run every day. 

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [x] API release notes have been updated if necessary
- [x] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)
- [x] Inform data insights team due to database changes
- [x] Make sure all information from the Trello card is in here
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally
- [x] Add PR link to Trello card

### CODE REVIEW

**Comment by elceebee** [comment]
[Bug: decline_by_default_at field is incorrectly being set](https://trello.com/c/2Vzt4gNQ/2315-bug-declinebydefaultat-field-is-incorrectly-being-set)

**Review by CatalinVoineag — APPROVED** [review]

File: config/analytics_blocklist.yml [hunk 1]
```diff
@@ -216,6 +216,8 @@
   :application_choices:
   - provider_ids
   - current_recruitment_cycle_year
+  - decline_by_default_at
```
Comment (dcyoung-dev) [inline config/analytics_blocklist.yml:219]: Have we communicated with analysts that these will be removed?

**Review by dcyoung-dev — APPROVED** [review]
Nice 👌

File: config/analytics_blocklist.yml (hunk 1 above)
Comment (elceebee) [reply config/analytics_blocklist.yml:219]: Yes, I've been chatting with Lewis about it. but I will let them know when it actually happens.

**Pull request merged**